src/test/groovy/org/prebid/server/functional/tests/SeatNonBidSpec.groovy

```diff
@@ -0,0 +1,251 @@
+            seatbid = []
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse, PBSUtils.getRandomElement(HttpStatusCode.values() as List))
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contain seatNonBid for called bidder"
+        assert response.ext.seatnonbid.size() == 1
+
+        def seatNonBid = response.ext.seatnonbid[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == FAILED_TO_REQUEST_BIDS
+    }
+
+    def "PBS should populate seatNonBid and debug when returnAllBidStatus=true, debug=0 and requested bidder didn't bid for any reason"() {
+        given: "Default bid request with returnAllBidStatus and debug"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = true
+            ext.prebid.debug = 0
+        }
+
+        and: "Default bidder response without bid"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid = []
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse, PBSUtils.getRandomElement(HttpStatusCode.values() as List))
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response should contain seatNonBid for called bidder"
+        assert response.ext.seatnonbid.size() == 1
+
+        def seatNonBid = response.ext.seatnonbid[0]
+        assert seatNonBid.seat == GENERIC.value
+        assert seatNonBid.nonBid[0].impId == bidRequest.imp[0].id
+        assert seatNonBid.nonBid[0].statusCode == FAILED_TO_REQUEST_BIDS
+
+        and: "PBS response shouldn't contain debug"
+        assert !response?.ext?.debug
+    }
+
+    def "PBS shouldn't populate seatNonBid when returnAllBidStatus=false and bidder didn't bid for any reason"() {
+        given: "Default bid request with disabled returnAllBidStatus"
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            ext.prebid.returnAllBidStatus = false
+        }
+
+        and: "Default bidder response without bids"
+        def bidResponse = BidResponse.getDefaultBidResponse(bidRequest).tap {
+            seatbid = []
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(bidRequest.id, bidResponse, PBSUtils.getRandomElement(HttpStatusCode.values() as List))
+
+        when: "PBS processes auction request"
+        def response = defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "PBS response shouldn't contain seatNonBid"
+        assert !response.ext.seatnonbid
+        assert !response.seatbid
+    }
+
+    def "PBS shouldn't populate seatNonBid when debug=0 and requested bidder didn't bid for any reason"() {
```

Comment:
This test can be squashed with the previous one by parametrizing `ext.prebid.debug`.

Reply:
You didn't squash it. You removed a perfectly valid test case. This feature works independently of the `debug` flag. Meaning that if it's disabled, PBS shouldn't emit non-bids independently of debug flag state. Meaning:
returnAllBidStatus=true:
- debug=0 == seatNonBid
- debug=1 == seatNonBid
- debug=null == seatNonBid

returnAllBidStatus=true:
- debug=0 == !seatNonBid
- debug=1 == !seatNonBid
- debug=null == !seatNonBid